tfe/data_source_variables.go

```diff
@@ -0,0 +1,124 @@
+package tfe
+
+import (
+	"fmt"
+	"log"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceTFEWorkspaceVariables() *schema.Resource {
+	varSchema := map[string]*schema.Schema{
+		"category": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"hcl": {
+			Type:     schema.TypeBool,
+			Computed: true,
+		},
+		"id": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"name": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"sensitive": {
+			Type:     schema.TypeBool,
+			Computed: true,
+		},
+		"value": {
+			Type:      schema.TypeString,
+			Computed:  true,
+			Sensitive: true,
+		},
+	}
+	return &schema.Resource{
+		Read: dataSourceVariableRead,
+
+		Schema: map[string]*schema.Schema{
```

Comment:
The data source works well! I think in the future we might want to add terraform_keys and env_keys as fields to enable other uses, but if this suits you for now it doesn't need to be added right away.

Reply:
@brandonc 
sorry the dealy the same day I did my last commit went to holidays 😄. It's great you have time to test it thanks for your help